prompterator/models/openai_models.py

```diff
@@ -86,12 +90,39 @@ def __init__(self):
         super().__init__()
 
     def call(self, idx, input, **kwargs):
+        structured_output_data: StructuredOutputData = kwargs["structured_output"]
         model_params = kwargs["model_params"]
         try:
+            if structured_output_data.enabled:
+                if structured_output_data.method == soi.FUNCTION_CALLING:
+                    model_params["tools"], function_name = build_function_calling_tooling(
```

Comment:
I'd argue that  functions with multiple return values are quite a nightmare 😅 Is there a way to perhaps split the function into two, each returning just one thing?

Reply:
I think these values are tightly coupled so it makes sense for me. Maybe in the whole context it is a bit "overwhelming" at first. I changed some other functions to make it more readable.